apps/tasktracker/services/task_tracker.go

```diff
@@ -13,3 +22,39 @@ func (service TaskTrackerService) ConfigList() (configList []map[string]interfac
 	}
 	return configList
 }
+
+// SupportedTimeTrackersList ...
+func (service TaskTrackerService) SupportedTimeTrackersList(taskTracker string, teamID string) (*serializers.TimeProvidersSerializer, error) {
+	var timeTrackerList serializers.TimeProvidersSerializer
+	var team userModels.Team
+	var isGenericTimeTracker bool
+
+	err := service.DB.Model(&userModels.Team{}).Where("id = ?", teamID).Scan(&team).Error
+	if err != nil {
+		return nil, err
+	}
+	// check if task tracker also provide time
+	if name, exists := timetracker.TimeProvidersDisplayNameMap[taskTracker]; exists {
+		timeTrackerList.TimeProviders = append(timeTrackerList.TimeProviders, serializers.TimeProvider{DisplayName: name, Name: taskTracker})
+	}
+
+	for _, genricTimetracker := range constants.GenericTimeTrackersList {
+		if team.TimeProviderName == genricTimetracker {
+			isGenericTimeTracker = true
+		} else {
+			timeTrackerList.TimeProviders = append(timeTrackerList.TimeProviders, serializers.TimeProvider{
+				Name:        genricTimetracker,
+				DisplayName: timetracker.TimeProvidersDisplayNameMap[genricTimetracker],
```

Comment:
Please check if it exceeds the line length as per the guidelines.

Reply:
no, it does not exceed line length.